### ISSUE TITLE
Custom filter selection?

### ISSUE THREAD

**Issue by petasis** [mistic100/jQuery-QueryBuilder#325]
Hi all,

Is it possible to have multiple comboboxes for selecting a filter, similar to the widgets demo, where a custom input is defined with two linked <select>?

I have a big list of filters, that I want to organise. optgroup can help, but I think it would be better if I can split the filter dropdown into 3 dropdowns.

The filters list I want to display is similar to this one:
http://ydsdev.iit.demokritos.gr/YDSComponents/#!/search-tabbed?tab=Subsidy
(click on "Show Advanced Search").

George

**Comment by mistic100** [comment]
#153

**Comment by mistic100** [comment]
See #153 this won't be implemented by default, you have to do it yourself

But what is the page you linked ? Isn't it yours ?

**Comment by petasis** [comment]
I am not sure I understand the question regarding the link. This link is an example for why a way to customise filters is needed. In the meantime, I have written a plugin that shows some menus.

http://ydsdev.iit.demokritos.gr/YDSComponents/#!/search-tabbed?tab=Subsidy

George

**Comment by mistic100** [comment]
My question was : is the page your linked yours ?

But you solved your problem right ? with the menus on the filter (I would be interested in your plugin BTW)

**Comment by petasis** [comment]
Yes, the page linked is mine. A colleague will provide more information about the plugin.

**Comment by leots** [comment]
Hello,
the plugin's code can be found here: https://github.com/YourDataStories/components-visualisation/blob/master/demo/lib/selectivity-querybuilder-plugin.js It uses [Selectivity](https://github.com/arendjr/selectivity) for the menu. But because we added submenus, it would probably be easier to understand [this version](https://github.com/YourDataStories/components-visualisation/blob/bcec096d013d251f2008672f083300ea0d045263/demo/lib/selectivity-querybuilder-plugin.js) which had a selectivity dropdown with no submenus and is more generic.

The way it works is that it creates a Selectivity dropdown and adds it next to the normal select element of Query Builder (in the rule filter container), then it hides QueryBuilder's select element with .hide().

It listens for Selectivity's "change" event which triggers when the user selects an option in Selectivity, and selects the appropriate option in the hidden <select> element, so QueryBuilder can see it. I also had to call .change() on the select element so that QueryBuilder actually realized that something changed and added the correct rule input and condition dropdowns.

Also when we give the Query Builder some starting rules, we have to manually select the correct option in Selectivity. This is done in the afterUpdateRuleFilter handler.

In the latest version with the submenus, because when the user selects an option in the Selectivity dropdowns, like beneficiary > refCode, the default behavior of Selectivity was to only show "refCode" as the selected filter, instead of "beneficiary refCode" like we wanted it to show. So I find the whole label, and edit Selectivity's span text to show it.

Leonidas
